runtime/parachains/src/runtime_api_impl/v1.rs

```diff
@@ -231,6 +232,29 @@ pub fn session_index_for_child<T: initializer::Config>() -> SessionIndex {
 	<shared::Module<T>>::session_index()
 }
 
+/// Implementation for the `AuthorityDiscoveryApi::authorities()` function of the runtime API.
+/// Gets current and historical authority ids using session_info module.
+pub fn get_current_and_historical_authority_ids<T: initializer::Config>() -> Vec<AuthorityDiscoveryId> {
+	let current_session_index = session_index_for_child::<T>();
+	let earliest_stored_session = <session_info::Module<T>>::earliest_stored_session();
+	let mut authority_ids = vec![];
+
+	// At session boundary, If there is a race condition between getting session indexes above and
+	// loop execution, we might miss latest session's authority ids, which can break the semantics.
```

Comment:
This code is not interruptible; I'm not sure what kind of race you're thinking would occur here.

Reply:
Good point about it being not interruptible. Is this always going to stay single-threaded and non-suspendable? I'd rather assume less.

Reply:
Interruptible is probably the wrong standard to use. Even if this code was interrupted, runtime APIs are always executed in the context of some state trie root hash and that would not change.